### PR TITLE
feat: add support for valkey

### PR DESCRIPTION
**Pull request by shadracnicholas**
- Intentionally update the storage size for the Percona database to 8Gi for dev and let users increase it in case they want to use it in prod.
- Fix the Percona database storageClass to enable default storage in the cluster.
- Fix worker config to enable passing in Redis, Valkey, or use the built-in option.

### CODE REVIEW

File: charts/shopware/Chart.lock [hunk 1]
```diff
@@ -5,6 +5,15 @@ dependencies:
...
   version: 0.0.13
+- name: valkey
+  repository: oci://registry-1.docker.io/bitnamicharts
+  version: 1.0.1
+- name: valkey
+  repository: oci://registry-1.docker.io/bitnamicharts
+  version: 1.0.1
+- name: valkey
+  repository: oci://registry-1.docker.io/bitnamicharts
+  version: 1.0.1
```
Comment (shyim) [inline charts/shopware/Chart.lock:16]: the three lines are the same looks wrong

File: charts/shopware/values.yaml [hunk 2]
```diff
@@ -31,8 +31,8 @@ percona:
     # annotations:
     # affinity:
     # serviceAnnotations:
-    # storageClassName:
-    # storageSize: 8Gi
+    # storageClassName: standard
```
Comment (TrayserCassa) [inline charts/shopware/values.yaml:34]: Why? The storage Class should be empty for the default storage Class.

https://kubernetes.io/docs/concepts/storage/persistent-volumes/#persistentvolumeclaims

File: charts/shopware/values.yaml [hunk 3]
```diff
@@ -31,8 +31,8 @@ percona:
...
+    storageSize: 8Gi
```
Comment (TrayserCassa) [inline charts/shopware/values.yaml:35]: Change the default in the helm chart and comment it here, as we do with all other values too

File: charts/shopware/Chart.lock (hunk 1 above)
Comment (renaudhager) [reply charts/shopware/Chart.lock:16]: Probably due to this:
```
  - name: valkey
    alias: valkeyapp
    version: 1.0.1
    repository: oci://registry-1.docker.io/bitnamicharts
    condition: valkeyapp.enabled

  - name: valkey
    alias: valkeysession
    version: 1.0.1
    repository: oci://registry-1.docker.io/bitnamicharts
    condition: valkeysession.enabled

  - name: valkey
    alias: valkeyworker
    version: 1.0.1
    repository: oci://registry-1.docker.io/bitnamicharts
    condition: valkeyworker.enabled
    ```

I guess Helm automatically updates the lock file for each chart imported.

File: charts/shopware/templates/database.yaml [hunk 4]
```diff
@@ -39,7 +39,7 @@ spec:
       {{- end }}
     volumeSpec:
       persistentVolumeClaim:
-        storageClassName: {{ .Values.percona.database.storageClassName | default "" | quote }}
+        storageClassName: {{ .Values.percona.database.storageClassName | quote }}
```
Comment (renaudhager) [inline charts/shopware/templates/database.yaml:42]: Is `storageClassName` required? If not, I think we should not set it and let Kubernetes use the default storage class.

File: charts/shopware/values.yaml [hunk 5]
```diff
@@ -257,9 +259,59 @@ redissession:
...
+        memory: 500Mi
+        cpu: 1000m
+      limits:
+        memory: 1Gi
+        cpu: 1000m
+  auth:
+    enabled: false
+
+valkeyworker:
+  enabled: false
```
Comment (TrayserCassa) [inline charts/shopware/values.yaml:279]: This should be enabled, right?

File: charts/shopware/values.yaml [hunk 6]
```diff
@@ -257,9 +259,59 @@ redissession:
...
+        cpu: 1000m
+      limits:
+        memory: 1Gi
+        cpu: 2000m
+  auth:
+    enabled: false
+
+redisapp:
+  enabled: false
+  architecture: standalone
```
Comment (TrayserCassa) [inline charts/shopware/values.yaml:296]: If we don't enable it, we can remove all other values. If somebody needs it and use it for now, they will have values for it.

File: charts/shopware/templates/store.yaml [hunk 7]
```diff
@@ -29,20 +29,39 @@ spec:
   {{- end }}
 
   sessionCache:
-    {{- if .Values.redissession.enabled }}
+    {{- if .Values.valkeysession.enabled }}
```
Comment (renaudhager) [inline charts/shopware/templates/store.yaml:32]: These two conditions are the same minus the value of the host. Would it be simpler if we use the condition and use a common `redisHost` variable that is set by a helper to either Valkey or Redis?

It seems `Redis` or `Valkey` use the "same" variable, so we could just use the as well and point them to either `Redis` or `Valkey`.

File: charts/shopware/templates/store.yaml [hunk 8]
```diff
@@ -29,20 +29,39 @@ spec:
...
+    {{- else if .Values.redissession.enabled }}
     adapter: redis
     redisHost: {{ template "getRedisSessionMasterService" . }}
     {{- else }}
     adapter: builtin
     {{- end }}
+
   appCache:
-    {{- if .Values.redisapp.enabled }}
+    {{- if .Values.valkeyapp.enabled }}
```
Comment (renaudhager) [inline charts/shopware/templates/store.yaml:43]: Same here.

File: charts/shopware/templates/store.yaml [hunk 9]
```diff
@@ -29,20 +29,39 @@ spec:
...
+    redisHost: {{ template "getValkeyAppMasterService" . }}
+    {{- else if .Values.redisapp.enabled }}
     adapter: redis
     redisHost: {{ template "getRedisAppMasterService" . }}
     {{- else }}
     adapter: builtin
     {{- end }}
 
+  worker:
+    {{- if .Values.valkeyworker.enabled }}
```
Comment (renaudhager) [inline charts/shopware/templates/store.yaml:54]: Same here.

File: charts/shopware/Chart.lock (hunk 1 above)
Comment (shadracnicholas) [reply charts/shopware/Chart.lock:16]: Its the lock file. We use aliases for the charts. Check redis.

File: charts/shopware/values.yaml (hunk 5 above)
Comment (shadracnicholas) [reply charts/shopware/values.yaml:279]: Yes

File: charts/shopware/templates/database.yaml (hunk 4 above)
Comment (TrayserCassa) [reply charts/shopware/templates/database.yaml:42]: That's done when you set it to `""`, I found that easier then do a check for the value. So just undo the change and done

**Review by renaudhager — APPROVED** [review]
LGTM.

**Pull request merged**